### ISSUE TITLE
What's the purpose of the project ?

### ISSUE THREAD

**Issue by okouyad** [Banno/kafka4s#639]
I just wonder given that there is fs2-kafka, what is the scope of this project? Is it just trying to give an alternative or does it try to address some ground not covered by fs2-kafka ?

**Comment by Kazark** [comment]
A piece of history: this project predates `fs2-kafka`. It takes a slightly different approach: rather than implementing the wire protocol, it wraps the Java client.

**Comment by okouyad** [comment]
Thank you for the prompt answer. I just discover your project and love the flexibility of it. The more I dig into the api, i just love it. This started because we have a use case where we need to produce to kafka with transaction, albeit not as part of consume-process-produce. We have strong mechanism to make sure that in case of failure, we resume from where we started,  but we needed transaction because we write to multiple topic at the same time and all those need to be atomic, also for idempotence of course. Akka stream do not support, fs2 that we have started to use since a year now, did not support the use  case either. But  API just make it super easy, and support the the exact use case. So thanks for this.
